Complete transition to std::span in CryptoDigest-related code

brentfulgham opened the pull request:
#### 779251105768188876c269d293475f2a65f2152c
<pre>
Complete transition to std::span in CryptoDigest-related code
<a href="https://bugs.webkit.org/show_bug.cgi?id=271488">https://bugs.webkit.org/show_bug.cgi?id=271488</a>
&lt;<a href="https://rdar.apple.com/problem/125256518">rdar://problem/125256518</a>&gt;

Reviewed by Chris Dumez and Sihui Liu.

Following the changes in Bug 249414 and Bug 271383, complete the work of
moving to std::span in CryptoDigest-related code.

* Source/JavaScriptCore/API/JSScript.mm:
(-[JSScript readCache]):
(-[JSScript writeCache:]):
* Source/JavaScriptCore/bytecode/CodeBlockHash.cpp:
(JSC::CodeBlockHash::CodeBlockHash):
* Source/JavaScriptCore/inspector/remote/glib/RemoteInspectorUtils.cpp:
(Inspector::backendCommandsHash):
* Source/JavaScriptCore/runtime/CachedBytecode.h:
(JSC::CachedBytecode::span const):
* Source/WTF/wtf/SHA1.h:
* Source/WTF/wtf/persistence/PersistentEncoder.cpp:
(WTF::Persistence::Encoder::updateChecksumForData):
* Source/WTF/wtf/persistence/PersistentEncoder.h:
(WTF::Persistence::Encoder::updateChecksumForNumber):
* Source/WebCore/Modules/webauthn/WebAuthenticationUtils.cpp:
(WebCore::buildClientDataJsonHash):
* Source/WebCore/Modules/websockets/WebSocketHandshake.cpp:
(WebCore::WebSocketHandshake::getExpectedWebSocketAccept):
* Source/WebCore/PAL/pal/crypto/CryptoDigest.h:
(PAL::CryptoDigest::addBytes): Deleted.
* Source/WebCore/PAL/pal/crypto/commoncrypto/CryptoDigestCommonCrypto.mm:
(PAL::CryptoDigest::addBytes):
* Source/WebCore/PAL/pal/crypto/gcrypt/CryptoDigestGCrypt.cpp:
(PAL::CryptoDigest::addBytes):
* Source/WebCore/PAL/pal/crypto/openssl/CryptoDigestOpenSSL.cpp:
(PAL::CryptoDigest::addBytes):
* Source/WebCore/PAL/pal/crypto/win/CryptoDigestWin.cpp:
(PAL::CryptoDigest::addBytes):
* Source/WebCore/css/FontFace.cpp:
(WebCore::FontFace::create):
* Source/WebCore/crypto/gcrypt/CryptoAlgorithmECDSAGCrypt.cpp:
(WebCore::gcryptSign):
(WebCore::gcryptVerify):
* Source/WebCore/crypto/gcrypt/CryptoAlgorithmRSASSA_PKCS1_v1_5GCrypt.cpp:
(WebCore::gcryptSign):
(WebCore::gcryptVerify):
* Source/WebCore/crypto/gcrypt/CryptoAlgorithmRSA_PSSGCrypt.cpp:
(WebCore::gcryptSign):
(WebCore::gcryptVerify):
* Source/WebCore/dom/Node.cpp:
(WebCore::hashPointer):
* Source/WebCore/inspector/DOMPatchSupport.cpp:
(WebCore::DOMPatchSupport::createDigest):
* Source/WebCore/loader/ResourceCryptographicDigest.cpp:
(WebCore::cryptographicDigestForBytes):
* Source/WebCore/loader/ResourceCryptographicDigest.h:
* Source/WebCore/loader/cache/TrustedFonts.cpp:
(WebCore::hashForFontData):
(WebCore::fontBinaryParsingPolicy):
* Source/WebCore/loader/cache/TrustedFonts.h:
* Source/WebCore/page/csp/ContentSecurityPolicy.cpp:
(WebCore::generateHashesForContent):
* Source/WebCore/platform/mediastream/RealtimeMediaSourceCenter.cpp:
(WebCore::addStringToSHA1):
* Source/WebCore/platform/network/soup/SoupNetworkSession.cpp:
(WebCore::HostTLSCertificateSet::computeCertificateHash):
* Source/WebCore/storage/StorageUtilities.cpp:
(WebCore::StorageUtilities::encodeSecurityOriginForFileName):
* Source/WebCore/workers/service/server/SWScriptStorage.cpp:
(WebCore::SWScriptStorage::sha2Hash const):
* Source/WebKit/NetworkProcess/cache/NetworkCacheData.cpp:
(WebKit::NetworkCache::computeSHA1):
* Source/WebKit/NetworkProcess/cache/NetworkCacheKey.cpp:
(WebKit::NetworkCache::hashString):
(WebKit::NetworkCache::Key::computeHash const):
(WebKit::NetworkCache::Key::partitionToPartitionHash):
* Source/WebKit/NetworkProcess/storage/CacheStorageDiskStore.cpp:
(WebKit::computeSHA1):
* Source/WebKit/NetworkProcess/storage/NetworkStorageManager.cpp:
(WebKit::encode):
* Source/WebKit/UIProcess/API/Cocoa/_WKWebAuthenticationPanel.mm:
(produceClientDataJsonHash):
(+[_WKWebAuthenticationPanel importLocalAuthenticatorWithAccessGroup:credential:error:]):
* Source/WebKit/UIProcess/WebAuthentication/Cocoa/LocalAuthenticator.mm:
(WebKit::LocalAuthenticator::continueMakeCredentialAfterUserVerification):
* Source/WebKit/UIProcess/WebAuthentication/Virtual/VirtualAuthenticatorUtils.mm:
(WebKit::credentialIdAndCosePubKeyForPrivateKey):
* Tools/DumpRenderTree/cg/PixelDumpSupportCG.cpp:
(computeSHA1HashStringForBitmapContext):
* Tools/TestWebKitAPI/Tests/WTF/BloomFilter.cpp:
(TestWebKitAPI::generateRandomDigests):
* Tools/TestWebKitAPI/Tests/WebCore/CryptoDigest.cpp:
(TestWebKitAPI::expect):
* Tools/TestWebKitAPI/cocoa/HTTPServer.mm:
(TestWebKitAPI::Connection::webSocketHandshake):
* Tools/WebKitTestRunner/cairo/TestInvocationCairo.cpp:
(WTR::computeSHA1HashStringForCairoSurface):
* Tools/WebKitTestRunner/skia/TestInvocationSkia.cpp:
(WTR::computeSHA1HashStringForPixmap):

* Tools/WebKitTestRunner/cg/TestInvocationCG.cpp:
(WTR::computeSHA1HashStringForContext):

Canonical link: <a href="https://commits.webkit.org/276673@main">https://commits.webkit.org/276673@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/20298cb81fc60e10ed238cc6b153e265465dd983

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/45278 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/24396 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/47798 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/47941 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/41286 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/47585 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/28592 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/21793 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/37133 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/45856 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/21451 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/39053 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/18233 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/18867 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/40138 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/3325 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/38498 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/41549 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/40455 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/49650 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/44747 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/20261 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/16793 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/44168 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/21567 "Built successfully") | | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/42981 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/10075 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/21928 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/51906 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/21254 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/10590 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/JavaScriptCore/API/JSScript.mm:
```diff
@@ -184,7 +184,7 @@ - (void)readCache
 
     SHA1::Digest computedHash;
     SHA1 sha1;
-    sha1.addBytes(fileData, fileDataSize);
```
Ideally, we'd be constructing the span as early as possible:
```
auto fileData = mappedFile.span();
// Ensure we at least have a SHA1::Digest to read.
if (fileData.size() < sizeof(SHA1::Digest)) {
    FileSystem::deleteFile(cacheFilename);
    return;
}
SHA1::Digest computedHash;
SHA1 sha1;
sha1.addBytes(fileData.left(fileData.size() - sizeof(SHA1::Digest)));
sha1.computeHash(computedHash);

auto hashSpan = dileData.right(sizeof(SHA1::Digest));
SHA1::Digest fileHash;
memcpy(&fileHash, hashSpan.data(), hashSpan.size());
```

brentfulgham replied:
Oh, that's indeed much cleaner. I'll switch to that approach.

szewai commented on Source/WebKit/NetworkProcess/cache/NetworkCacheKey.cpp:
```diff
@@ -89,20 +89,20 @@ static void hashString(SHA1& sha1, const String& string)
     auto cString = string.utf8();
     // Include terminating null byte.
-    sha1.addBytes(cString.dataAsUInt8Ptr(), cString.length() + 1);
+    sha1.addBytes(std::span { cString.dataAsUInt8Ptr(), cString.length() + 1 });
```
`spanIncludingNullTerminator`?

brentfulgham replied:
Yes! Thank you.

cdumez commented on Source/WebCore/Modules/websockets/WebSocketHandshake.cpp:
```diff
@@ -104,7 +104,7 @@ String WebSocketHandshake::getExpectedWebSocketAccept(const String& secWebSocket
     CString keyData = secWebSocketKey.ascii();
     sha1.addBytes(keyData.span());
-    sha1.addBytes(webSocketKeyGUID, std::size(webSocketKeyGUID) - 1);
+    sha1.addBytes(std::span { webSocketKeyGUID, std::size(webSocketKeyGUID) - 1 });
```
Maybe we can use this above:
```
constexpr auto webSocketKeyGUID = "258EAFA5-E914-47DA-95CA-C5AB0DC85B11"_span;
```
And then just pass `webSocketKeyGUID ` to `addBytes()` here.

brentfulgham replied:
Oh, cool! I didn't know that was an option. I'll try that.

brentfulgham replied:
I couldn't figure out how to make this work without compile errors. I tried numerous approaches, but in the end largely left this unchanged.

brentfulgham replied:
I don't know what I did wrong before. I tried again and it compiled properly. Weird.

szewai commented on Source/WebCore/platform/mediastream/RealtimeMediaSourceCenter.cpp:
```diff
@@ -157,12 +157,12 @@ static void addStringToSHA1(SHA1& sha1, const String& string)
 
     auto utf8 = string.utf8();
-    sha1.addBytes(utf8.dataAsUInt8Ptr(), utf8.length() + 1); // Include terminating null byte.
+    sha1.addBytes(std::span { utf8.dataAsUInt8Ptr(), utf8.length() + 1 }); // Include terminating null byte.
```
`spanIncludingNullTerminator`?

cdumez commented on Source/WebCore/PAL/pal/crypto/openssl/CryptoDigestOpenSSL.cpp:
```diff
@@ -131,10 +131,10 @@ std::unique_ptr<CryptoDigest> CryptoDigest::create(CryptoDigest::Algorithm algor
 {
     ASSERT(m_context);
-    m_context->addBytes(input, length);
+    m_context->addBytes(static_cast<const void*>(input.data()), input.size());
```
Would be nice to update this function to take in a std::span so we can just pass `input` here.

brentfulgham replied:
Oh, yes. Just another layer of the onion. I can make that change, too.

cdumez replied:
Weird, this builds just fine for me:
```
diff --git a/Source/WebCore/Modules/websockets/WebSocketHandshake.cpp b/Source/WebCore/Modules/websockets/WebSocketHandshake.cpp
index b82d5f48b051..540241109707 100644
--- a/Source/WebCore/Modules/websockets/WebSocketHandshake.cpp
+++ b/Source/WebCore/Modules/websockets/WebSocketHandshake.cpp
@@ -100,11 +100,11 @@ static String generateSecWebSocketKey()
 
 String WebSocketHandshake::getExpectedWebSocketAccept(const String& secWebSocketKey)
 {
-    constexpr uint8_t webSocketKeyGUID[] = "258EAFA5-E914-47DA-95CA-C5AB0DC85B11";
+    const auto webSocketKeyGUID = "258EAFA5-E914-47DA-95CA-C5AB0DC85B11"_span;
     SHA1 sha1;
     CString keyData = secWebSocketKey.ascii();
     sha1.addBytes(keyData.span());
-    sha1.addBytes(webSocketKeyGUID, std::size(webSocketKeyGUID) - 1);
+    sha1.addBytes(webSocketKeyGUID);
     SHA1::Digest hash;
     sha1.computeHash(hash);
     return base64EncodeToString(hash.data(), SHA1::hashSize);
```

cdumez replied:
So does this:
```
diff --git a/Source/WebCore/Modules/websockets/WebSocketHandshake.cpp b/Source/WebCore/Modules/websockets/WebSocketHandshake.cpp
index b82d5f48b051..425aecf22ed8 100644
--- a/Source/WebCore/Modules/websockets/WebSocketHandshake.cpp
+++ b/Source/WebCore/Modules/websockets/WebSocketHandshake.cpp
@@ -100,11 +100,11 @@ static String generateSecWebSocketKey()
 
 String WebSocketHandshake::getExpectedWebSocketAccept(const String& secWebSocketKey)
 {
-    constexpr uint8_t webSocketKeyGUID[] = "258EAFA5-E914-47DA-95CA-C5AB0DC85B11";
+    constexpr auto webSocketKeyGUID = "258EAFA5-E914-47DA-95CA-C5AB0DC85B11"_s;
     SHA1 sha1;
     CString keyData = secWebSocketKey.ascii();
     sha1.addBytes(keyData.span());
-    sha1.addBytes(webSocketKeyGUID, std::size(webSocketKeyGUID) - 1);
+    sha1.addBytes(webSocketKeyGUID.span8());
     SHA1::Digest hash;
     sha1.computeHash(hash);
     return base64EncodeToString(hash.data(), SHA1::hashSize);
```

Either would be better IMO.

brentfulgham replied:
I'll try again. Maybe I had a typo or something. I'm running up-to-date Xcode, but I can double-check versions there if it continues to fail.

cdumez commented on Source/WebCore/PAL/pal/crypto/win/CryptoDigestWin.cpp:
```diff
@@ -80,11 +80,11 @@ std::unique_ptr<CryptoDigest> CryptoDigest::create(Algorithm algorithm)
+void CryptoDigest::addBytes(std::span<const uint8_t> input)
 {
-    if (!input || !length)
+    if (!input.data() || !input.size())
```
`if (input.empty())` should suffice here.

brentfulgham replied:
Will fix.

szewai commented on Source/WebCore/PAL/pal/crypto/openssl/CryptoDigestOpenSSL.cpp:
```diff
@@ -131,10 +131,10 @@ std::unique_ptr<CryptoDigest> CryptoDigest::create(CryptoDigest::Algorithm algor
 {
     ASSERT(m_context);
-    m_context->addBytes(input, length);
+    m_context->addBytes(static_cast<const void*>(input.data()), input.size());
```
It seems addBytes should take a span

cdumez commented on Source/WebCore/platform/mediastream/RealtimeMediaSourceCenter.cpp:
```diff
@@ -157,12 +157,12 @@ static void addStringToSHA1(SHA1& sha1, const String& string)
 
     auto utf8 = string.utf8();
-    sha1.addBytes(utf8.dataAsUInt8Ptr(), utf8.length() + 1); // Include terminating null byte.
+    sha1.addBytes(std::span { utf8.dataAsUInt8Ptr(), utf8.length() + 1 }); // Include terminating null byte.
```
`sha1.addBytes(string.utf8().spanIncludingNullTerminator());`

No need for the `utf8` local anymore either.

brentfulgham replied:
Nice!

cdumez commented on Source/WebCore/storage/StorageUtilities.cpp:
```diff
@@ -93,7 +93,7 @@ String encodeSecurityOriginForFileName(FileSystem::Salt salt, const SecurityOrig
     auto originString = origin.toString().utf8();
     crypto->addBytes(originString.span());
-    crypto->addBytes(salt.data(), salt.size());
+    crypto->addBytes(std::span { salt });
```
Do we really need the explicit `std::span { }` here? Maybe we can just pass `salt`?

brentfulgham replied:
That seems to work, indeed!

cdumez commented on Source/WebCore/workers/service/server/SWScriptStorage.cpp:
```diff
@@ -51,7 +51,7 @@ SWScriptStorage::SWScriptStorage(const String& directory)
 {
     auto crypto = PAL::CryptoDigest::create(PAL::CryptoDigest::Algorithm::SHA_256);
-    crypto->addBytes(m_salt.data(), m_salt.size());
+    crypto->addBytes(std::span { m_salt });
```
ditto.

cdumez commented on Source/WebKit/NetworkProcess/cache/NetworkCacheData.cpp:
```diff
@@ -82,7 +82,7 @@ Data adoptAndMapFile(FileSystem::PlatformFileHandle handle, size_t offset, size_
 {
     SHA1 sha1;
-    sha1.addBytes(salt.data(), salt.size());
+    sha1.addBytes(std::span { salt });
```
ditto.

cdumez commented on Source/WebKit/NetworkProcess/cache/NetworkCacheKey.cpp:
```diff
@@ -89,20 +89,20 @@ static void hashString(SHA1& sha1, const String& string)
     auto cString = string.utf8();
     // Include terminating null byte.
-    sha1.addBytes(cString.dataAsUInt8Ptr(), cString.length() + 1);
+    sha1.addBytes(std::span { cString.dataAsUInt8Ptr(), cString.length() + 1 });
```
Same comment as earlier, no need for the `cString` local and should rely on `Cstring::spanIncludingNullTerminator()`

brentfulgham replied:
Will fix.

cdumez commented on Source/WebKit/NetworkProcess/cache/NetworkCacheKey.cpp:
```diff
@@ -89,20 +89,20 @@ static void hashString(SHA1& sha1, const String& string)
     // SHA1 just happens to be suitably sized, fast and available.
     SHA1 sha1;
-    sha1.addBytes(salt.data(), salt.size());
+    sha1.addBytes(std::span { salt });
```
is `std::span { }` really needed?

brentfulgham replied:
I assumed it was needed to go from `std::array` -> `std::span`, but I'll try it without.

cdumez commented on Source/WebKit/NetworkProcess/cache/NetworkCacheKey.cpp:
```diff
@@ -127,7 +127,7 @@ Key::HashType Key::computePartitionHash(const Salt& salt) const
 {
     SHA1 sha1;
-    sha1.addBytes(salt.data(), salt.size());
+    sha1.addBytes(std::span { salt });
```
ditto.

cdumez commented on Source/WebKit/NetworkProcess/storage/CacheStorageDiskStore.cpp:
```diff
@@ -54,7 +54,7 @@ static bool shouldStoreBodyAsBlob(const Vector<uint8_t>& bodyData)
 {
     SHA1 sha1;
-    sha1.addBytes(salt.data(), salt.size());
+    sha1.addBytes(std::span { salt });
```
ditto.

cdumez commented on Source/WebKit/NetworkProcess/storage/NetworkStorageManager.cpp:
```diff
@@ -83,7 +83,7 @@ static String encode(const String& string, FileSystem::Salt salt)
     auto utf8String = string.utf8();
     crypto->addBytes(utf8String.span());
-    crypto->addBytes(salt.data(), salt.size());
+    crypto->addBytes(std::span { salt });
```
ditto.

cdumez reviewed (approved):
LGTM with improvement ideas.

szewai commented on Source/WebCore/Modules/websockets/WebSocketHandshake.cpp:
```diff
@@ -104,7 +104,7 @@ String WebSocketHandshake::getExpectedWebSocketAccept(const String& secWebSocket
     CString keyData = secWebSocketKey.ascii();
     sha1.addBytes(keyData.span());
-    sha1.addBytes(webSocketKeyGUID, std::size(webSocketKeyGUID) - 1);
+    sha1.addBytes(std::span { webSocketKeyGUID, std::size(webSocketKeyGUID) - 1 });
```
webSocketKeyGUID should be `std::array`, and we can call `span()` on it.

brentfulgham replied:
Will fix, thank you!

brentfulgham replied:
I couldn't get this to cleanly convert to `std::array` (or directly to `std::span`). The type inference makes this an array of `char[37]`, and `std::array` (or `span`) want you to pass it the start and length. There's probably a clever way to do this but I couldn't figure it out quickly.

cdumez replied:
`std::to_array<uint8_t>()` may help. But it's OK to leave as is for now if you can't make it work.

brentfulgham committed:
Complete transition to std::span in CryptoDigest-related code
https://bugs.webkit.org/show_bug.cgi?id=271488
<rdar://problem/125256518>

Reviewed by Chris Dumez and Sihui Liu.

Following the changes in Bug 249414 and Bug 271383, complete the work of
moving to std::span in CryptoDigest-related code.

* Source/JavaScriptCore/API/JSScript.mm:
(-[JSScript readCache]):
(-[JSScript writeCache:]):
* Source/JavaScriptCore/bytecode/CodeBlockHash.cpp:
(JSC::CodeBlockHash::CodeBlockHash):
* Source/JavaScriptCore/inspector/remote/glib/RemoteInspectorUtils.cpp:
(Inspector::backendCommandsHash):
* Source/JavaScriptCore/runtime/CachedBytecode.h:
(JSC::CachedBytecode::span const):
* Source/WTF/wtf/SHA1.h:
* Source/WTF/wtf/persistence/PersistentEncoder.cpp:
(WTF::Persistence::Encoder::updateChecksumForData):
* Source/WTF/wtf/persistence/PersistentEncoder.h:
(WTF::Persistence::Encoder::updateChecksumForNumber):
* Source/WebCore/Modules/webauthn/WebAuthenticationUtils.cpp:
(WebCore::buildClientDataJsonHash):
* Source/WebCore/Modules/websockets/WebSocketHandshake.cpp:
(WebCore::WebSocketHandshake::getExpectedWebSocketAccept):
* Source/WebCore/PAL/pal/crypto/CryptoDigest.h:
(PAL::CryptoDigest::addBytes): Deleted.
* Source/WebCore/PAL/pal/crypto/commoncrypto/CryptoDigestCommonCrypto.mm:
(PAL::CryptoDigest::addBytes):
* Source/WebCore/PAL/pal/crypto/gcrypt/CryptoDigestGCrypt.cpp:
(PAL::CryptoDigest::addBytes):
* Source/WebCore/PAL/pal/crypto/openssl/CryptoDigestOpenSSL.cpp:
(PAL::CryptoDigest::addBytes):
* Source/WebCore/PAL/pal/crypto/win/CryptoDigestWin.cpp:
(PAL::CryptoDigest::addBytes):
* Source/WebCore/css/FontFace.cpp:
(WebCore::FontFace::create):
* Source/WebCore/crypto/gcrypt/CryptoAlgorithmECDSAGCrypt.cpp:
(WebCore::gcryptSign):
(WebCore::gcryptVerify):
* Source/WebCore/crypto/gcrypt/CryptoAlgorithmRSASSA_PKCS1_v1_5GCrypt.cpp:
(WebCore::gcryptSign):
(WebCore::gcryptVerify):
* Source/WebCore/crypto/gcrypt/CryptoAlgorithmRSA_PSSGCrypt.cpp:
(WebCore::gcryptSign):
(WebCore::gcryptVerify):
* Source/WebCore/dom/Node.cpp:
(WebCore::hashPointer):
* Source/WebCore/inspector/DOMPatchSupport.cpp:
(WebCore::DOMPatchSupport::createDigest):
* Source/WebCore/loader/ResourceCryptographicDigest.cpp:
(WebCore::cryptographicDigestForBytes):
* Source/WebCore/loader/ResourceCryptographicDigest.h:
* Source/WebCore/loader/cache/TrustedFonts.cpp:
(WebCore::hashForFontData):
(WebCore::fontBinaryParsingPolicy):
* Source/WebCore/loader/cache/TrustedFonts.h:
* Source/WebCore/page/csp/ContentSecurityPolicy.cpp:
(WebCore::generateHashesForContent):
* Source/WebCore/platform/mediastream/RealtimeMediaSourceCenter.cpp:
(WebCore::addStringToSHA1):
* Source/WebCore/platform/network/soup/SoupNetworkSession.cpp:
(WebCore::HostTLSCertificateSet::computeCertificateHash):
* Source/WebCore/storage/StorageUtilities.cpp:
(WebCore::StorageUtilities::encodeSecurityOriginForFileName):
* Source/WebCore/workers/service/server/SWScriptStorage.cpp:
(WebCore::SWScriptStorage::sha2Hash const):
* Source/WebKit/NetworkProcess/cache/NetworkCacheData.cpp:
(WebKit::NetworkCache::computeSHA1):
* Source/WebKit/NetworkProcess/cache/NetworkCacheKey.cpp:
(WebKit::NetworkCache::hashString):
(WebKit::NetworkCache::Key::computeHash const):
(WebKit::NetworkCache::Key::partitionToPartitionHash):
* Source/WebKit/NetworkProcess/storage/CacheStorageDiskStore.cpp:
(WebKit::computeSHA1):
* Source/WebKit/NetworkProcess/storage/NetworkStorageManager.cpp:
(WebKit::encode):
* Source/WebKit/UIProcess/API/Cocoa/_WKWebAuthenticationPanel.mm:
(produceClientDataJsonHash):
(+[_WKWebAuthenticationPanel importLocalAuthenticatorWithAccessGroup:credential:error:]):
* Source/WebKit/UIProcess/WebAuthentication/Cocoa/LocalAuthenticator.mm:
(WebKit::LocalAuthenticator::continueMakeCredentialAfterUserVerification):
* Source/WebKit/UIProcess/WebAuthentication/Virtual/VirtualAuthenticatorUtils.mm:
(WebKit::credentialIdAndCosePubKeyForPrivateKey):
* Tools/DumpRenderTree/cg/PixelDumpSupportCG.cpp:
(computeSHA1HashStringForBitmapContext):
* Tools/TestWebKitAPI/Tests/WTF/BloomFilter.cpp:
(TestWebKitAPI::generateRandomDigests):
* Tools/TestWebKitAPI/Tests/WebCore/CryptoDigest.cpp:
(TestWebKitAPI::expect):
* Tools/TestWebKitAPI/cocoa/HTTPServer.mm:
(TestWebKitAPI::Connection::webSocketHandshake):
* Tools/WebKitTestRunner/cairo/TestInvocationCairo.cpp:
(WTR::computeSHA1HashStringForCairoSurface):
* Tools/WebKitTestRunner/skia/TestInvocationSkia.cpp:
(WTR::computeSHA1HashStringForPixmap):

* Tools/WebKitTestRunner/cg/TestInvocationCG.cpp:
(WTR::computeSHA1HashStringForContext):

Canonical link: https://commits.webkit.org/276673@main
